File: src/main/java/edu/cit/estillore/MentorMatch/MentorMatch/common/ApiError.java
package edu.cit.estillore.MentorMatch.MentorMatch.common;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Map;

@Data
@AllArgsConstructor
public class ApiError {
    private String message;
    private Map<String, String> fieldErrors; // null when not a field-validation error
}

File: android/app/src/main/java/edu/cit/estillore/mentormatch/data/model/ApiError.kt
package edu.cit.estillore.mentormatch.data.model

import com.squareup.moshi.JsonClass

/**
 * Mirrors ApiError.java. fieldErrors is populated for @Valid failures
 * (e.g. registration form field errors), null for simple message-only errors
 * such as duplicate email or bad credentials.
 */
@JsonClass(generateAdapter = true)
data class ApiError(
    val message: String?,
    val fieldErrors: Map<String, String>? = null
)
